Kentor.AuthServices/Internal/PendingAuthInMemoryContainer.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Kentor.AuthServices.Internal
+{
+    /// <summary>
+    /// In memory container for storing in-progress Authentication requests. 
+    /// </summary>
+    internal static class PendingAuthnInMemoryContainer
```

Comment:
Remove this class. It doesn't make sense to have a class that just wraps a static dictionary. Better to go all the way and have an instance as part of IOptions or ISPOptions and get rid of `static` completely. That will make the isolation better if multiple instances of the Owin middleware are used in the same process too.